### ISSUE TITLE
Changes to Cr dimensions in SMFB_ES

### ISSUE THREAD

**Issue by pdolder** [flr/FLBEIA#34]
Hi,

When I try the gravity.flbeia function I get the error:

```
Error in Cr[stknm, ] : incorrect number of dimensions
```
Has something changed?

Thanks,
Paul

**Comment by dorleta** [comment]
Hi Paul,

Yes, we introduced the Baranov production function last week and in fact the changes affected to Cr. I’ll check the code and see if I can be more helpful,

Dorleta


De: Paul Dolder <notifications@github.com>
Enviado el: Friday, 8 May 2020 17:33
Para: flr/FLBEIA <FLBEIA@noreply.github.com>
CC: Subscribed <subscribed@noreply.github.com>
Asunto: [flr/FLBEIA] Changes to Cr dimensions in SMFB_ES (#34)


Hi,

When I try the gravity.flbeia function I get the error:

Error in Cr[stknm, ] : incorrect number of dimensions

Has something changed?

Thanks,
Paul

**Comment by pdolder** [comment]
Ok, sound interesting! I'll await to hear more details and if you point to
the problem I can fix as needed.

Paul

**Comment by dorleta** [comment]
Ok I’m working this weekend so I’ll try look at it before monday

De: Paul Dolder <notifications@github.com>
Enviado el: Friday, 8 May 2020 17:49
Para: flr/FLBEIA <FLBEIA@noreply.github.com>
CC: Dorleta Garcia <dgarcia@azti.es>; Comment <comment@noreply.github.com>
Asunto: Re: [flr/FLBEIA] Changes to Cr dimensions in SMFB_ES (#34)

Ok, sound interesting! I'll await to hear more details and if you point to
the problem I can fix as needed.

Paul

**Comment by dorleta** [comment]
Hi Paul,

I’ve been looking at the code and I think the best way to proceed is to take the new SMFB function code exactly as it is now, and in line 103 before the stocks loop, paste the lines 233-241 in your code, so the two functions are exactly the same in all the code but the update of the effort share. I’m not completely sure if the change will solve the problem completely, but we will limit it to the call of “effortShare.fun” function. Cr.f, the object which dimension has changed, has dimension [stks, it], but it seems that it is not used in the gravity and the other models, so it shouldn’t be problematic.

Cheers,

Dorleta

De: Paul Dolder <notifications@github.com>
Enviado el: Friday, 8 May 2020 17:49
Para: flr/FLBEIA <FLBEIA@noreply.github.com>
CC: Dorleta Garcia <dgarcia@azti.es>; Comment <comment@noreply.github.com>
Asunto: Re: [flr/FLBEIA] Changes to Cr dimensions in SMFB_ES (#34)

Ok, sound interesting! I'll await to hear more details and if you point to
the problem I can fix as needed.

Paul

**Comment by pdolder** [comment]
Hi Dorleta,

Thanks for taking a look at this quickly. I'll test out what you suggest
and let you know if it works.

Paul

**Comment by pdolder** [comment]
Hi,

I have made the changes and alongside some other small tweaks; as N is now passed as an array, not an flq, it required a few reference changes in the effort share models.

I have tested and all seems ok so made a pull request.

Only issue is in the previous version we passed N and B by recalculating mid-season,...but now the N and B are from the start of the season. I don't think this is a huge problem.

Paul
